pkgs/top-level/haskell-packages.nix

```
@@ -349,6 +351,22 @@ in {
       buildTargetLlvmPackages = pkgsBuildTarget.llvmPackages_12;
       llvmPackages = pkgs.llvmPackages_12;
     };
+    ghc946 = callPackage ../development/compilers/ghc/9.4.6.nix {
+      bootPkgs =
+        # Building with 9.2 is broken due to
+        # https://gitlab.haskell.org/ghc/ghc/-/issues/21914
+        # Use 8.10 as a workaround where possible to keep bootstrap path short.
+
+        # On ARM text won't build with GHC 8.10.*
+        if stdenv.hostPlatform.isAarch then
+          # TODO(@sternenseemann): package bindist
+          packages.ghc902
+        # No suitable bindists for powerpc64le
+        else if stdenv.hostPlatform.isPower64 && stdenv.hostPlatform.isLittleEndian then
+          packages.ghc902
+        else
+          packages.ghc8107Binary;
```

Thanks for backporting this!

It looks like you're getting an eval error in ofborg on darwin.

My guess is that you at least need to add the following line from the above 9.4.5:

```suggestion
          packages.ghc8107Binary;
      # Need to use apple's patched xattr until
      # https://github.com/xattr/xattr/issues/44 and
      # https://github.com/xattr/xattr/issues/55 are solved.
      inherit (buildPackages.darwin) xattr autoSignDarwinBinariesHook;
```

And possibly the two lines about LLVM as well.

---

Also, when doing backports, make sure you use the `git cherry-pick -x` command, so that the commit message for the cherry-pick links to the original commit.

---

I'll amend to add the message with the original commit (maybe after I see if this ofborg eval succeeds...)